app/models/project_tag.rb

```diff
@@ -0,0 +1,9 @@
+class ProjectTag < ApplicationRecord
+  # --- Associations --- #
+  belongs_to :project_log
+  belongs_to :tag
+
+  # --- Validations --- #
+  validates :project_log_id, :tag_id, presence: true
```

Comment:
@mjllewelyn Your issue is here. You have a `:presence` validation on `:project_log_id`. You can remove both presence validations on the _id fields since the `belongs_to` validates the association presence by default.